eng/versioning.targets

```diff
@@ -0,0 +1,82 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <GenerateAssemblyInfo Condition="'$(GenerateAssemblyInfo)'==''">true</GenerateAssemblyInfo>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(GenerateAssemblyInfo)'=='true'">
+    <AssemblyInfoFile Condition="'$(MSBuildProjectExtension)' == '.csproj'">$(IntermediateOutputPath)_AssemblyInfo.cs</AssemblyInfoFile>
+    <AssemblyInfoFile Condition="'$(MSBuildProjectExtension)' == '.vbproj'">$(IntermediateOutputPath)_AssemblyInfo.vb</AssemblyInfoFile>
+  </PropertyGroup>
+
+
+  <Target Name="DecideIfWeNeedToIncludeDllSafeSearchPathAttribute"
+    Condition="'$(IsDotNetFrameworkProductAssembly)' == 'true' AND '$(IsTestProject)' != 'true'">
+
+    <!-- We want to apply the IncludeDllSafeSearchPathAttribute on all DotNet assemblies (non test) that have a reference to System.Runtime.InteropServices -->
```

Comment:
Is this new behavior?  What about assemblies referencing netstandard or CoreLib?  Those *could* have it as well.  You could probably simplify this by scheduling it after ResolveReferences and before CoreCompile and just examining ReferencePath items instead.

Reply:
No this is existing behavior I just copied from BuildTools https://github.com/dotnet/buildtools/blob/master/src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets#L43. We could probably clean this up but for now it is doing the job.

Reply:
I see.  I did a quick check of the shared framework and didn't notice anything that was missing this attribute but also using DllImport, so I guess we're ok.